src/Kafka/Consumer.hs

```diff
@@ -1,8 +1,58 @@
 {-# LANGUAGE LambdaCase        #-}
 {-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE TupleSections     #-}
+
+-----------------------------------------------------------------------------
+-- |
+-- Module to consume messages from Kafka topics.
+-- 
+-- Here's an example of code to consume messages from a topic:
+-- 
+-- @
+-- import Control.Exception (bracket)
+-- import Data.Monoid ((<>))
+-- import Kafka
+-- import Kafka.Consumer
+-- 
+-- -- Global consumer properties
+-- consumerProps :: 'ConsumerProperties'
```

Comment:
I reused a technique I found in Aeson documentation where code examples also link all their types/functions. It's more verbose in the Haddock, but it provides links to everything provided by this library, which I think is a great way to discover and understand it:
![image](https://user-images.githubusercontent.com/1204125/80006972-e7c39600-84c5-11ea-957c-d6be98a3c2be.png)